src/utils.go

```diff
@@ -105,19 +135,61 @@ func QuerySmartContractData(nodeUrl string, contractAddress string,
 			"status_code": resp.StatusCode,
 			"response":    string(body),
 		})
+
+		// Try to parse error message
+		var wasmErr WasmError
+		if err := json.Unmarshal(body, &wasmErr); err == nil {
+			return nil, fmt.Errorf("error fetching smart contract data - wasm error response: %v", wasmErr.Message)
+		}
 		return nil, fmt.Errorf("fetching smart contract data: %d", resp.StatusCode)
 	}
 
-	var response map[string]interface{}
+	var response struct {
```

Comment:
needed to change the return type here since some responses just didn't fit it, e.g. a single number as data